mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -39,6 +53,61 @@ mlir::Attribute addElemToArrayAttr(mlir::PatternRewriter &rewriter,
   values.push_back(element);
   return rewriter.getArrayAttr(values);
 }
+
+LogicalResult add(mlir::PatternRewriter &rewriter, mlir::PDLResultList &results,
+                  llvm::ArrayRef<mlir::PDLValue> args) {
+  assert(args.size() == 2 && "Expected 2 arguments");
+  auto lhsAttr = args[0].cast<Attribute>();
+  auto rhsAttr = args[1].cast<Attribute>();
+
+  // Integer
+  if (auto lhsIntAttr = dyn_cast_or_null<IntegerAttr>(lhsAttr)) {
```

Comment:
@mgehre-amd does it make sense to use `rewriter->getInsertionPoint()` and generate a meaningful error that both operands must be IntergerAttr or FloatAttr? or even without the insertion point because right now, it's just failing and we don't know why it's failing.

Reply:
In the design used in this PR, this is a match failure, not an error. So we shouldn't generate an error message. This might flood the output with messages. 
Note that the design will potentially change depending on our decision and reception in the open source community, so we might need to rewrite this.